checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1708 @@
+   * the size of its resource alias set), etc.
+   *
+   * @param node the assignment
+   * @param obligations the set of obligations to update
+   */
+  private void handleAssignment(
+      AssignmentNode node, Set<ImmutableSet<LocalVarWithTree>> obligations) {
+    Node lhs = node.getTarget();
+    Element lhsElement = TreeUtils.elementFromTree(lhs.getTree());
+    // Use the temporary variable for the rhs if it exists.
+    Node rhs = removeCasts(node.getExpression());
+    LocalVariableNode tempVarForRhs = typeFactory.getTempVarForNode(rhs);
+    if (tempVarForRhs != null) {
+      rhs = tempVarForRhs;
+    }
+
+    // Ownership transfer to @Owning field.
+    if (lhsElement.getKind() == ElementKind.FIELD) {
+      boolean isOwningField =
+          !checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)
+              && typeFactory.getDeclAnnotation(lhsElement, Owning.class) != null;
+      // Check that there is no obligation on the lhs, if the field is non-final and owning.
+      if (isOwningField
+          && typeFactory.canCreateObligations()
+          && !ElementUtils.isFinal(lhsElement)) {
+        checkReassignmentToField(node, obligations);
+      }
+      // Remove obligations from local variables, now that the owning field is responsible.
+      // (When obligation creation is turned off, non-final fields cannot take ownership.)
+      if (isOwningField
+          && rhs instanceof LocalVariableNode
+          && (typeFactory.canCreateObligations() || ElementUtils.isFinal(lhsElement))) {
+        removeObligationContainingVar(obligations, (LocalVariableNode) rhs);
+      }
+    } else if (lhs instanceof LocalVariableNode) {
+      LocalVariableNode lhsVar = (LocalVariableNode) lhs;
+      doGenKillForPseudoAssignment(node, obligations, lhsVar, rhs);
+    }
+  }
+
+  /**
+   * Remove any obligations that contain {@code var} in their resource-alias set.
+   *
+   * @param obligations the set of obligations
+   * @param var a variable
+   */
+  private void removeObligationContainingVar(
+      Set<ImmutableSet<LocalVarWithTree>> obligations, LocalVariableNode var) {
+    Set<LocalVarWithTree> setContainingRhs = getResourceAliasSetForVar(obligations, var);
+    obligations.remove(setContainingRhs);
+  }
+
+  /**
+   * Remove any {@link TypeCastNode}s wrapping a node, returning the operand nested within the type
+   * casts.
+   *
+   * @param node a node
+   * @return node, but with any surrounding typecasts removed
+   */
+  private Node removeCasts(Node node) {
+    while (node instanceof TypeCastNode) {
+      node = ((TypeCastNode) node).getOperand();
+    }
+    return node;
+  }
+
+  /**
+   * Update a set of tracked obligations to account for a (pseudo-)assignment to some variable, as
+   * in a gen-kill dataflow analysis problem. Pseudo-assignments may include operations that
+   * "assign" to a temporary variable. E.g., for an expression {@code b ? x : y}, this method may
```

Comment:
The examples are useful, but please state the principle as well.  Are these "pass-through" or "no-op" operations?

Reply:
Maybe @msridhar can answer this better than I can. I think these are exactly the same expressions for which a temporary variable is created (i.e. those listed in the class Javadoc), but I'm not 100% sure.

Reply:
We use temporary variables to simulate a three-address-form IR.  In such an IR, you could imagine a ternary expression `b ? x : y` with corresponding temp. var. `t` looking something like:
```
if (b) then t = x else t = y
```
These assignments `t = x` and `t = y` are what the Javadoc is discussing.  It's not so much about being a "no-op" operation as it is capturing the semantic meaning of the temp. var. for a ternary expression correctly.  I can try to tweak the docs to capture this

Reply:
e5a189c45f2e21a819e7d7c2e704dbbece08964b

Reply:
Thanks for incorporating that example.  I think the fact that these emulate three-address-form IR is helpful too and should appear in the documentation.  That answers my questions about whether a temp var is created for every subexpression of an expression (the answer is "yes").

Reply:
See 3bd3b4d41488456f924593394a8c6e8307f83b53 for an additional mention of three-address-form IR